staging/src/k8s.io/cli-runtime/pkg/genericclioptions/config_flags.go

```diff
@@ -439,13 +447,21 @@ func NewConfigFlags(usePersistentConfig bool) *ConfigFlags {
 		// given 25 groups (our groups + a few custom resources) with one-ish version each, discovery needs to make 50 requests
 		// double it just so we don't end up here again for a while.  This config is only used for discovery.
 		discoveryBurst: 100,
+		CacheTTL:       durationPtr(1 * time.Hour),
```

Comment:
In my opinion the default can even be a full day. CRDs don't change that often and if a GVK can't be found in the cache, the cache is rendered invalid and repopulated anyway.

Reply:
I think, let cache-ttl default to 1 hour and we also have a configuration option for the same. 1 day looks very long duration.